### PR TITLE
Add search guide

### PR DESCRIPTION
**Pull request by mooreds**
This reworks some of the search documentation and adds a substantial new guide which walks people through how to search using the Elasticsearch search engine in FusionAuth.

### CODE REVIEW

File: site/docs/v1/tech/admin-guide/_troubleshooting-elasticsearch-queries.adoc [hunk 1]
```diff
@@ -0,0 +1,76 @@
+When you are troubleshooting Elasticsearch queries, running them against the Elasticsearch server is helpful. If you are running in FusionAuth Cloud, however, you cannot access Elasticsearch.
+
+When running FusionAuth locally in Docker, you can expose the Elasticsearch port by adding the following lines to your `docker-compose.yml` file and restart.
```
Comment (andrewpai) [inline site/docs/v1/tech/admin-guide/_troubleshooting-elasticsearch-queries.adoc:3]:
```suggestion
When running FusionAuth locally in Docker, you can expose the Elasticsearch port by adding the following lines to your `docker-compose.yml` file and restarting.
```

File: site/docs/v1/tech/admin-guide/_troubleshooting-elasticsearch-queries.adoc [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+curl -X GET "localhost:9200/_search?pretty" -H 'Content-Type: application/json' -d' 
+{
+  "query": {
+    "match": {
+      "data.Company": {
+        "query": "PiedPiper"
+      }
+    }
+  }
+}
```
Comment (andrewpai) [inline site/docs/v1/tech/admin-guide/_troubleshooting-elasticsearch-queries.adoc:29]:
```suggestion
}'
```

File: site/docs/v1/tech/guides/user-search-with-elasticsearch.adoc [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+== User Search Examples
+
+Below are examples of searches you can run. All examples are available to be run locally by downloading https://github.com/fusionauth/fusionauth-example-user-search[this GitHub repository] and following the instructions in the README.
+
+Each example uses the below shell script to run the search. Feel free to download it, update your API key and base FusionAuth URL, and experiment. The script uses the `POST` method so that the request body can be large. The value of the filename passed to the script changes each time, but everything else is the same. 
+
+[source,shell] 
+.`search.sh`: An Example Shell Script For Searching
+----
+include::https://raw.githubusercontent.com/FusionAuth/fusionauth-example-user-search/main/examples/search.sh[]
```
Comment (andrewpai) [inline site/docs/v1/tech/guides/user-search-with-elasticsearch.adoc:119]: When this is rendered, the #!/bin/sh is hardly visible. Is that easily changed?

File: site/docs/v1/tech/admin-guide/_troubleshooting-elasticsearch-queries.adoc [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      {
+        "_index" : "fusionauth_user",
+        "_type" : "_doc",
+        "_id" : "00000000-0000-0000-0000-000000000005",
+        "_score" : 2.0
+      }
+    ]
+  }
+}
+---
```
Comment (andrewpai) [inline site/docs/v1/tech/admin-guide/_troubleshooting-elasticsearch-queries.adoc:73]: These dashes are getting included in the formatted JSON block. Is that intentional?

**Review by andrewpai — APPROVED** [review]

File: site/docs/v1/tech/admin-guide/_troubleshooting-elasticsearch-queries.adoc (hunk 4 above)
Comment (mooreds) [reply site/docs/v1/tech/admin-guide/_troubleshooting-elasticsearch-queries.adoc:73]: nope. Will take a look.

File: site/docs/v1/tech/guides/user-search-with-elasticsearch.adoc (hunk 3 above)
Comment (mooreds) [reply site/docs/v1/tech/guides/user-search-with-elasticsearch.adoc:119]: Only took a major version upgrade of the build tool :)

**Pull request merged**